src/http/request.rs

```diff
@@ -104,6 +115,21 @@ impl Request {
         unsafe { Pool::from_ngx_pool(self.0.pool) }
     }
 
+    /// Returns the result as an `Option` if it exists, otherwise `None`.
+    ///
+    /// The option wraps an ngx_http_upstream_t instance, it will be none when the underlying NGINX request
+    /// does not have a pointer to a [`ngx_http_upstream_t`] upstream structure.
+    ///
+    /// [`ngx_http_upstream_t`]: is best described in
+    /// https://nginx.org/en/docs/dev/development_guide.html#http_request
+    /// https://nginx.org/en/docs/dev/development_guide.html#http_load_balancing
+    pub fn upstream(&self) -> Option<*mut ngx_http_upstream_t> {
```

Comment:
nit: 
totally optionally we can start using some [name conventions](https://rust-lang.github.io/api-guidelines/naming.html)  (I know there is no such convention as `_raw_mut` on the link, but why not?)

`pub fn upstream_raw_mut(&self) -> Option<*mut ngx_http_upstream_t> {`

similarly to `_mut` naming convention, from the rust recommendation::

```
pub struct S {
    first: First,
    second: Second,
}

impl S {
    // Not get_first.
    pub fn first(&self) -> &First {
        &self.first
    }

    // Not get_first_mut, get_mut_first, or mut_first.
    pub fn first_mut(&mut self) -> &mut First {
        &mut self.first
    }
}
```

Reply:
A lot of Request doesn't follow a consistent pattern - I'll present a diff to discuss.

Reply:
This will come in a following commit. We've decided to consistently name all accessors and functions.